### ISSUE TITLE
Number input doesn't update data in some circumstances.

### ISSUE THREAD

**Issue by trvrm** [ractivejs/ractive#2761]
## Description:

If the first keypress I do as a user on a input control of type `number` empties
the number field, then that change is not reflected in the underlying data.

This only occurs if there is a single digit value in the control, and the very first thing I do is delete that value.  If I otherwise modify it by typing, using the arrow keys, or clicking on the number control spinners, then the underlying data updates correctly.
 
## Versions affected:

Ractive 0.8.3

## Platforms affected:

Chrome

## Reproduction:

http://codepen.io/trvrm/pen/pyOeyR

To observe this behaviour the first action you do on loading this pen *must* be to delete the number from the number control.  If you take any other actions, you may need to refresh the page to see this error.

## **Code:**

```coffeescript
ractive = new Ractive     
    el: '#target'
    template: '''
        <input type="number" value={{x}} />
        <hr>
        <pre>x is {{x}}</pre>
    '''
    data:
        x:7 
```

**Comment by ghost** [comment]
Just to clarify: The problem is when you delete (backspace, delete, cut) the value of an `<input type="number">` when the data provided is a single digit, the data doesn't get updated (the data is still that one digit).

**Comment by trvrm** [comment]
Yes. I've come up with a workaround at http://codepen.io/trvrm/pen/eBJovy.  If I manually reset the underlying value after the control has been rendered, the problem goes away.

**Comment by ghost** [comment]
~~Hmm... Seems like the number input is thinking that its last value (before emptying) was `undefined`.~~ Actually, it should be undefined. It's just that numeric inputs return `undefined` when empty. A check is failing somewhere. 🤔

**Comment by evs-chris** [comment]
Yep. The last value tracking isn't being initialized correctly, and numerics happen to use undefined as their empty value. The fix for this will go out in 0.8.5. Thanks for the report!

**Comment by ghost** [comment]
> The last value tracking isn't being initialized correctly

Ahh, so that explains why `<input type="text">` also returned `undefined` as last value. It just worked because the check was `undefined === ''`.
